testing/matrix-sdk-test/Cargo.toml

```diff
@@ -18,13 +18,22 @@ doctest = false
 [dependencies]
 as_variant = { workspace = true }
 http = { workspace = true }
+insta = { workspace = true }
 matrix-sdk-common = { path = "../../crates/matrix-sdk-common" }
 matrix-sdk-test-macros = { version = "0.7.0", path = "../matrix-sdk-test-macros" }
 once_cell = { workspace = true }
 # Enabling the unstable feature for polls support.
-ruma = { workspace = true, features = ["rand", "unstable-msc3381"] }
+ruma = { workspace = true, features = ["canonical-json", "rand", "unstable-msc3381"] }
 serde = { workspace = true }
-serde_json = { workspace = true }
+
+# We enable the `preserve_order` feature in serde_json so that json snapshots are consistent.
+# The problem is that otherwise, we get different results depending on whether the whole workspace is built at once
+# (in which case we pull in some of the MAS crates such as `mas-oidc-client`, which enable `preserve_order`) or
+# building an individual crate (in which case `preserve_order` may not be defined, and JSON snapshots are generated
+# with keys sorted alphabetically).
+serde_json = { workspace = true, features = ["preserve_order"] }
```

Comment:
Previously we solved this with the `insta` setting: https://docs.rs/insta/latest/insta/struct.Settings.html#method.sort_maps.

Does this not work for you?

Reply:
ahhhh, I thought something like that must exist, but failed to find it.

Reply:
I've rebased, to include this change, so that we can still see the evolution of the snapshots over the lifetime of the PR.